pkg/netutils/netutils_test.go

```diff
@@ -0,0 +1,1026 @@
+// Copyright (c) 2019 Multus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package netutils
+
+import (
+	"encoding/json"
+	"net"
+	"testing"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types/020"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/containernetworking/plugins/pkg/testutils"
+
+	"github.com/vishvananda/netlink"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestNetutils(t *testing.T) {
+    RegisterFailHandler(Fail)
+    RunSpecs(t, "netutils")
+}
+
+// helper function
+func testAddRoute(link netlink.Link, ip net.IP, mask net.IPMask, gw net.IP) error {
+    dst := &net.IPNet{
+        IP:   ip,
+        Mask: mask,
+    }
+    route := netlink.Route{LinkIndex: link.Attrs().Index, Dst: dst, Gw: gw}
+    return netlink.RouteAdd(&route)
+}
+
+func testAddAddr(link netlink.Link, ip net.IP, mask net.IPMask) error {
+    return netlink.AddrAdd(link, &netlink.Addr{IPNet: &net.IPNet{IP: ip, Mask: mask}})
+}
+
+func testGetResultFromCache(data []byte) []byte {
+	var cachedInfo map[string]interface{}
+	err := json.Unmarshal(data, &cachedInfo)
+	ExpectWithOffset(1,err).NotTo(HaveOccurred())
```

Comment:
nit: you could inline these:
```golang
	ExpectWithOffset(1, json.Unmarshal(data, &cachedInfo)).NotTo(HaveOccurred())
```

I'd again use `Expect(...).To(Succeed())` rather than `Expect(...).NotTo(HaveOccurred())` but this is a personal opinion; you're free to Ignore both suggestions if you prefer.